test/Riok.Mapperly.Tests/Mapping/ObjectPropertyConstructorResolverTest.cs

```diff
@@ -349,5 +349,37 @@ public void RecordToRecordFlattenedNonDirectAssignmentNullHandling()
             .Should()
             .HaveSingleMethodBody(@"var target = new B(source.Nested == null ? throw new System.ArgumentNullException(nameof(source.Nested.Value)) : (double)source.Nested.Value);
     return target;");
+    }
+
+    [Fact]
+    public void CanResolveToRecordConstructorWithMapPropertyAttribute()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            "[MapProperty(nameof(ThingClass.Id), nameof(ThingRecord.ThingId))] partial ThingRecord ToRecord(ThingClass thing);",
```

Comment:
This codebase almost always uses mapping test classes A and B, therefore I'd rename `ThingClass` to `A` and `ThingRecord` to `B` to be consistent.